src/Symfony/Component/Config/Resource/DirectoryResource.php

```diff
@@ -87,7 +87,12 @@ public function isFresh($timestamp)
                 continue;
             }
 
-            $newestMTime = max($file->getMTime(), $newestMTime);
+            // early return if any file mtime is greater than passed timestamp
+            if (($fileMTime = $file->getMTime()) > $timestamp) {
+                return false;
+            }
+
+            $newestMTime = max($fileMTime, $newestMTime);
```

Comment:
It's rather a bug fix to me. 
For reference, [`FileResource`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Config/Resource/FileResource.php#L65) uses `filemtime($this->resource) <= $timestamp` too.